File: LeetCode/Simulation/JS/Leetcode1678-goal-parser-interpretation.js
var interpret = function(s) {
  let [res, n] = [[], s.length];
  for(let i = 0;i < n;i++) {
      if(s[i] == 'G') res.push('G');
      else {
          let j = i;
          while(s[j] != ')') j++;
          if(j - i <= 1) {
              res.push('o');
          } else {
              res.push("al");
          }
          i = j;
      }
  }
  return res.join('');
};

File: LeetCode/Simulation/Python/Leetcode1678-goal-parser-interpretation.py
class Solution:
    def interpret(self, s: str) -> str:
        res,i, n = [], 0, len(s)
        while i < n:
            if s[i] == 'G': res.append(s[i])
            else:
                j = i
                while s[j] != ')': j += 1
                if j - i <= 1: res.append('o')
                else: res.append('al')
                i = j
            i += 1
        return ''.join(res)